### ISSUE TITLE
'app' of undefined

### ISSUE THREAD

**Issue by ClementLoridan** [faceyspacey/webpack-flush-chunks#34]
Hi, 

I have this message in my terminal:

```
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'app' of undefined
```

```
const chunkNames = flushChunkNames();
const { js, styles, cssHash } = flushChunks(assets, { chunkNames, before: [], after: ['app'] });
```

I'm doing something wrong?

Thk

**Comment by faceyspacey** [comment]
Too general my friend. Keep digging. Isolate the problem. Do that by making your code more similar to the examples. Find what's different about your code from the example. Good luck :)

Log stats.assetsByChunkName
Log chunkNames
Search all the GitHub issues; chances are people have run into the exact same issues

**Comment by ClementLoridan** [comment]
Ok thk, and one question: it's possible to render a component sync by server and an other component async by the client in the component sync?

Example : Render Header, sidebar sync and render the content by client

Thk

**Comment by faceyspacey** [comment]
i dont know what you mean by:

- "in the component sync"
- "render content by client"

But yes, the whole purpose is to render things synchronously on the server, and syncronously on the client when possible, and *fallback* to rendering asynchronously. 

If what you're asking is whether you can have some stuff render async and other stuff sync, it's not really geared towards that use case, but it is possibly by simply not serving the corresponding chunks to the client. React 16 now doesn't complain when it can't render the whole tree, and will instead just patch the different parts. 

Now that said, the component will render synchronously on the server anyway. So the question is really how to block that? You could simply guard it with `{typeof window !== 'undefined' && <MyAsyncComponent />}`.

You could also PR the react-universal-component repo and add an option called `asyncOnly` and do the same in there by doing a similar check in a maybe 2-4 places.

**Comment by ClementLoridan** [comment]
Bonjour,

This is my render:

```
class Home extends Component {

  render() {
    return (
      <div>
          {
              this.props.isConnected ?
                  <HeaderConnected />
              :   <HeaderNoConnected />
          }
          <Feed />
          <Sidebar />
      </div>
    );
  }
}
```

I would like, render my Home with:

`The correct Header` (corresponding Js & Css) => Server rendering

`Sidebar` => server rendering

`Feed` or (content) => by the client: (Why : because I can receive item every second)

Your idea is `asyncOnly` would be suitable for that? 🤔

**Comment by faceyspacey** [comment]
Yes, but u should make sure the server also renders the loading... spinner. Same as the client on first load. Traditionally there would be "checksum mismatches." It's less or a problem with 16.

So rendering a spinner of equivalent on the server and on first load on the client, and then on the client only render once you have the data.

This is more of a data issue than an async import issue. 

Also I think you can just add a custom resolve option to the universal hoc that returns null or an empty string for the module to weakly an synchronously require. Then it simply will never render any other way than async. This is your new asyncOnly option :)

**Comment by ClementLoridan** [comment]
Ok, I see asyncOnly in `index`, and I see to pass this in option. But there I am hard to follow, is render Home (with universal) and render in the Home, an other universal component.

```
import React from 'react'
import styles from '../css/Foo'

const Header = universal(props => import('./Header'), {
  minDelay: 1200,
  loading: Loading,
  error: NotFound
})

export default () =>
  <div className={styles.container}>
  <Header />
    <span>Foo -- loaded</span>
  </div>
```

**Comment by faceyspacey** [comment]
I honestly I can't understand you well brother. Based on your use case you don't need to do what you want to do. Just use this hoc the way it's meant to be used and don't render anything in the given component until you get data on the client.

**Comment by ClementLoridan** [comment]
With this, do you understand?

*Home:* (when user is connected)
<link rel="stylesheet" href="/static/Home.css">
<link rel="stylesheet" href="/static/HeaderConnected.css">
<link rel="stylesheet" href="/static/Feed.css">
<link rel="stylesheet" href="/static/Sidebar.css">
[...]
<script type="text/javascript" src="/static/Home.js"></script>
<script type="text/javascript" src="/static/HeaderConnected.js"></script>
<script type="text/javascript" src="/static/Feed.js"></script>
<script type="text/javascript" src="/static/Sidebar.js"></script>

*In Foo page:* (when user is not connected)
<link rel="stylesheet" href="/static/Foo.css">
<link rel="stylesheet" href="/static/HeaderNoConnected.css">
<link rel="stylesheet" href="/static/Feed.css">
[...]
<script type="text/javascript" src="/static/Foo.js"></script>
<script type="text/javascript" src="/static/HeaderNoConnected.js"></script>
<script type="text/javascript" src="/static/Feed.js"></script>

Feed is content in Home and Foo. (maybe I'm lost 😭 :p )

**Comment by faceyspacey** [comment]
I don't know what connected is. I'm on codementor.io if you would like to pay for support. This after all is an issue tracker. Ie for bugs. I try to help as much as I can. But I don't know what you mean by the language you are using, ie "connected"

I'm on my phone. I'm gonna have to sign off until Monday. Maybe by then you can present your case more clearly. I'm sorry my friend.

**Comment by ClementLoridan** [comment]
Connected is when my user sign-in or sign-up
